daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/infoset/JsonInfosetInputter.scala

```diff
@@ -172,7 +180,13 @@ class JsonInfosetInputter(input: java.io.InputStream) extends InfosetInputter {
             objectDepth -= 1; exitNow = true
 
           // start of a simple type or null
-          case JsonToken.VALUE_STRING | JsonToken.VALUE_NULL => exitNow = true
+          case JsonToken.VALUE_STRING |
+               JsonToken.VALUE_NUMBER_INT |
+               JsonToken.VALUE_NUMBER_FLOAT |
+               JsonToken.VALUE_TRUE |
+               JsonToken.VALUE_FALSE |
+               JsonToken.VALUE_NULL =>
+            exitNow = true
```

Comment:
It looks like there is a `JsonToken.isScalar()` function that I think returns true if it's one of these VALUE_* tokens. So this case can probably be simplified to `case token if token.isScalar => exitNow = true`. Need to double check that's the case.